presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduBucketFunction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kudu;
+
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.connector.BucketFunction;
+import io.prestosql.spi.type.Type;
+import org.apache.kudu.Schema;
+import org.apache.kudu.client.KeyEncoderAccessor;
+import org.apache.kudu.client.KuduTable;
+import org.apache.kudu.client.PartialRow;
+import org.apache.kudu.client.PartitionSchema;
+import org.apache.kudu.client.PartitionSchema.HashBucketSchema;
+
+import java.util.List;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+
+public class KuduBucketFunction
+        implements BucketFunction
+{
+    private final int bucketCount;
+    private final List<Integer> bucketColumnIdIndexes;
+    private final List<Type> bucketChannelTypes;
+
+    private final List<HashBucketSchema> hashBucketSchemas;
+    private final Schema schema;
+
+    KuduBucketFunction(int bucketCount, List<Integer> bucketColumnIdIndexes, List<Type> bucketChannelTypes, KuduTable kuduTable)
+    {
+        this.bucketCount = bucketCount;
+        this.bucketColumnIdIndexes = bucketColumnIdIndexes;
+        this.bucketChannelTypes = bucketChannelTypes;
+
+        PartitionSchema partitionSchema = kuduTable.getPartitionSchema();
+        this.hashBucketSchemas = partitionSchema.getHashBucketSchemas();
+        this.schema = kuduTable.getSchema();
+    }
+
+    @Override
+    public int getBucket(Page page, int position)
+    {
+        return getBucketId(bucketChannelTypes, page, position);
+    }
+
+    private int getBucketId(List<Type> types, Page page, int position)
+    {
+        PartialRow partialRow = new PartialRow(schema);
+        List<Integer> bucketIds = this.hashBucketSchemas.stream().map(hashBucketSchema -> {
+            for (int i = 0; i < page.getChannelCount(); i++) {
```

Comment:
We need to populate only the row for bucketed columns right ?

Reply:
Here `page` is generated by presto contains only bucketed columns.

Reply:
![Screenshot from 2020-06-30 11-37-11](https://user-images.githubusercontent.com/33892550/86080915-97207880-bac6-11ea-93b7-29519e9afe69.png)
![Screenshot from 2020-06-30 11-33-30](https://user-images.githubusercontent.com/33892550/86080923-9e478680-bac6-11ea-8d30-8bf776b28cd0.png)